### ISSUE TITLE
Question about softmax

### ISSUE THREAD

**Issue by jameSssssss** [ashwhall/dsnt#10]
I think the _softmax2d should implent as ：
tf.exp(logits) / tf.reduce_sum(tf.exp(logits),[1,2], keepdims=True).

Why subtract the input by max value first as in dsnt.py  _softmax2d as below? 
max_axis = tf.reduce_max(target, axes, keepdims=True)
target_exp = tf.exp(target - max_axis)
